commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java

```diff
@@ -0,0 +1,171 @@
+        Path frontendDir = moduleDir.toPath().resolve(Path.of("src", "main", FRONTEND_FOLDER_NAME));
+        if (Files.isDirectory(frontendDir)) {
+            this.frontendFolder = frontendDir;
+            LOGGER.debug("Using frontend folder {}", frontendDir);
+        } else if (Files.isDirectory(legacyFrontendFolder)) {
+            this.frontendFolder = legacyFrontendFolder;
+            LOGGER.debug("Using legacy frontend folder {}", legacyFrontendFolder);
+        } else {
+            LOGGER.debug("Frontend folder not detected");
+        }
+    }
+
+    @Override
+    public Path getInputDirectory() {
+        return frontendFolder;
+    }
+
+    @Override
+    public boolean shouldRun(Path sourceDir, Config config) {
+        if (!Files.isDirectory(sourceDir)) {
+            return false;
+        }
+        String prefix = computeConnectClientPrefix(config);
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = sourceDir.resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            try {
+                String content = Files.readString(customClient);
+                if (!content.contains("const client = new ConnectClient({prefix: '" + prefix + "'});")) {
+                    LOGGER.debug(
+                            "Custom connect-client.ts detected ({}), but prefix does not match configuration {}.",
+                            customClient,
+                            prefix);
+                    return true;
+                }
+            } catch (IOException e) {
+                LOGGER.debug("Custom connect-client.ts detected ({}), but cannot read content.", customClient);
+                return false;
+            }
+            LOGGER.debug("Custom connect-client.ts detected ({}) with expected prefix {}.", customClient, prefix);
+            return false;
+        } else if (!defaultPrefix) {
+            LOGGER.debug("Custom prefix {} detected, connect-client.ts to be created in {}.", prefix, customClient);
+        }
+        return !defaultPrefix;
+    }
+
+    @Override
+    public boolean trigger(CodeGenContext context) {
+        String prefix = computeConnectClientPrefix(context.config());
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = context.inputDir().resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            String content = null;
+            try {
+                content = Files.readString(customClient);
+            } catch (IOException e) {
+                LOGGER.debug(
+                        "Cannot read content of custom connect-client.ts ({}). File will not be overwritten.",
+                        customClient);
+            }
+            if (content != null) {
+                content = content.replaceFirst(
+                        "(const client = new ConnectClient\\(\\{prefix:\\s*')[^']+('}\\);)", "$1" + prefix + "$2");
```

Comment:
Would only filter starting for `new ConnectClient(...)` (without `;`), a custom client could use another naming or add some trailing stuff in the line.

Reply:
Maybe also define the regex somewhere static

Reply:
If the client doesn't match the template I would not make any change